### PR TITLE
Implement for_window support for dim_inactive

### LINKED ISSUE #68: Ignore dim on certain windows
With `dim_inactive`, some windows get dimmed that I would like to not be dimmed, like my browser for example. I would propose that you can use `for_window` to set the dim for certain windows. For example:

`for_window [app_id="LibreWolf"] dim_inactive 0`

Experimenting with this, I found a bug. When I set the `for_window` like this, and I open a new tab, `dim_inactive` gets ignored globally until I reload sway. So I am not sure if this is a bug or not implemented yet.

### PR DESCRIPTION
**Pull request by famfo**
Note: this still requires more testing, I just did some basic tests on my setup so far.

This should fix #68.

### CODE REVIEW

**Comment by famfo** [comment]
I have tested this for a bit in my day to day stuff, didn't find any issues, so I am marking this as ready for review.

File: sway/commands/dim_inactive.c [hunk 1]
```diff
@@ -18,6 +18,10 @@ struct cmd_results *cmd_dim_inactive(int argc, char **argv) {
 	}
 
 	struct sway_container *container = config->handler_context.container;
+    if (!container) {
+        return cmd_results_new(CMD_INVALID, "cmd_dim connot be used without a for_window rule");
```
Comment (head-gardener) [inline sway/commands/dim_inactive.c:22]: cannot* typo

File: sway/meson.build [hunk 2]
```diff
@@ -52,9 +52,10 @@ sway_sources = files(
 	'commands/smart_corner_radius.c',
 	'commands/create_output.c',
 	'commands/default_border.c',
+	'commands/default_dim_inactive.c',
 	'commands/default_floating_border.c',
 	'commands/default_orientation.c',
-	'commands/dim_inactive.c',
+    'commands/dim_inactive.c',
```
Comment (WillPower3309) [inline sway/meson.build:58]: Nit: match spacing

**Review by WillPower3309 — CHANGES_REQUESTED** [review]
Apologies for the late review, love the work you've done here! Minor NIT

File: sway/meson.build (hunk 2 above)
Comment (famfo) [reply sway/meson.build]: My editor and git have different ideas of how may tabs to put in that file :)
Fixed in the latest commit.

File: sway/desktop/render.c [hunk 3]
```diff
@@ -1186,7 +1191,9 @@ static void render_containers_stacked(struct sway_output *output,
 			.dim_color = view_is_urgent(current->view)
 							 ? config->dim_inactive_colors.urgent
 							 : config->dim_inactive_colors.unfocused,
-			.dim = current->current.focused || parent->focused ? 0.0f: config->dim_inactive,
+			.dim = current->current.focused || parent->focused 
+                ? 0.0f
+                : current->dim,
```
Comment (WillPower3309) [inline sway/desktop/render.c:1196]: One line

File: sway/desktop/render.c [hunk 4]
```diff
@@ -1286,7 +1293,9 @@ static void render_floating_container(struct sway_output *soutput,
 			.dim_color = view_is_urgent(view)
 							 ? config->dim_inactive_colors.urgent
 							 : config->dim_inactive_colors.unfocused,
-			.dim = con->current.focused ? 0.0f: config->dim_inactive,
+			.dim = con->current.focused 
+                ? 0.0f
+                : con->dim,
```
Comment (WillPower3309) [inline sway/desktop/render.c:1298]: One line

File: sway/desktop/render.c [hunk 5]
```diff
@@ -1477,7 +1486,9 @@ void output_render(struct sway_output *output, struct timespec *when,
 			.dim_color = view_is_urgent(focus->view)
 				 ? config->dim_inactive_colors.urgent
 				 : config->dim_inactive_colors.unfocused,
-			.dim = focus->current.focused ? 0.0f: config->dim_inactive,
+			.dim = focus->current.focused 
+                ? 0.0f
+                : focus->dim,
```
Comment (WillPower3309) [inline sway/desktop/render.c:1491]: One line

File: sway/tree/container.c [hunk 6]
```diff
@@ -42,6 +42,7 @@ struct sway_container *container_create(struct sway_view *view) {
 	c->view = view;
 	c->alpha = 1.0f;
 	c->saturation = 1.0f;
+    c->dim = config->default_dim_inactive;
```
Comment (WillPower3309) [inline sway/tree/container.c:45]: Nit: Match indentation

**Review by WillPower3309 — CHANGES_REQUESTED** [review]
More nits, once those are addressed I think it's good to merge

**Comment by famfo** [comment]
Fixed some other indention issues as well, everything should be properly indented now now.

**Pull request merged**